codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieDecoder.java

```diff
@@ -32,47 +32,61 @@
     private final InternalLogger logger = InternalLoggerFactory.getInstance(getClass());
 
     private final boolean strict;
+    private final boolean reportException;
+    private final boolean needReport;
+
+    protected CookieDecoder(boolean strict, boolean reportException) {
+       this.strict = strict;
+       this.reportException = reportException;
+       needReport = reportException || logger.isDebugEnabled();
+    }
 
     protected CookieDecoder(boolean strict) {
-        this.strict = strict;
+        this(strict, false);
     }
 
     protected DefaultCookie initCookie(String header, int nameBegin, int nameEnd, int valueBegin, int valueEnd) {
         if (nameBegin == -1 || nameBegin == nameEnd) {
-            logger.debug("Skipping cookie with null name");
+            if (needReport) {
+                reportError(String.format("Invalid cookie with null name found in header '%s'", header));
```

Comment:
@Scottmitch something like this:
```java
...
reportError(() -> String.format("Invalid cookie with null name found in header '%s'", header));
...

private void reportError(Supplier<String> messageSupplier) {
    if (reportException) {
        throw new IllegalArgumentException(messageSupplier.get());
    }
    if (logger.isDebugEnabled()) {
        logger.debug(messageSupplier.get());
    }
}
```

Such approach (lazy string computation through lambdas) is used e.g. in the `log4j2` [1].

[1] https://logging.apache.org/log4j/2.x/manual/api.html#LambdaSupport